src/ppx/grfx/grfx_format.cpp

```diff
@@ -18,208 +18,168 @@
-    // FORMAT_BC1_RGB_SRGB,
-    {FORMAT_DATA_TYPE_SRGB, FORMAT_ASPECT_COLOR, 2, -1, FORMAT_LAYOUT_COMPRESSED, FORMAT_COMPONENT_RED_GREEN_BLUE, {{-1, -1, -1, -1}}},
-    // FORMAT_BC1_RGB_UNORM,
-    {FORMAT_DATA_TYPE_UNORM, FORMAT_ASPECT_COLOR, 2, -1, FORMAT_LAYOUT_COMPRESSED, FORMAT_COMPONENT_RED_GREEN_BLUE, {{-1, -1, -1, -1}}},
-    // FORMAT_BC2_SRGB,
-    {FORMAT_DATA_TYPE_SRGB, FORMAT_ASPECT_COLOR, 4, -1, FORMAT_LAYOUT_COMPRESSED, FORMAT_COMPONENT_RED_GREEN_BLUE_ALPHA, {{-1, -1, -1, -1}}},
-    // FORMAT_BC2_UNORM,
-    {FORMAT_DATA_TYPE_UNORM, FORMAT_ASPECT_COLOR, 4, -1, FORMAT_LAYOUT_COMPRESSED, FORMAT_COMPONENT_RED_GREEN_BLUE_ALPHA, {{-1, -1, -1, -1}}},
-    // FORMAT_BC3_SRGB,
-    {FORMAT_DATA_TYPE_SRGB, FORMAT_ASPECT_COLOR, 4, -1, FORMAT_LAYOUT_COMPRESSED, FORMAT_COMPONENT_RED_GREEN_BLUE_ALPHA, {{-1, -1, -1, -1}}},
-    // FORMAT_BC3_UNORM,
-    {FORMAT_DATA_TYPE_UNORM, FORMAT_ASPECT_COLOR, 4, -1, FORMAT_LAYOUT_COMPRESSED, FORMAT_COMPONENT_RED_GREEN_BLUE_ALPHA, {{-1, -1, -1, -1}}},
-    // FORMAT_BC4_UNORM,
-    {FORMAT_DATA_TYPE_UNORM, FORMAT_ASPECT_COLOR, 2, -1, FORMAT_LAYOUT_COMPRESSED, FORMAT_COMPONENT_RED, {{-1, -1, -1, -1}}},
-    // FORMAT_BC4_SNORM,
-    {FORMAT_DATA_TYPE_SNORM, FORMAT_ASPECT_COLOR, 2, -1, FORMAT_LAYOUT_COMPRESSED, FORMAT_COMPONENT_RED, {{-1, -1, -1, -1}}},
-    // FORMAT_BC5_UNORM,
-    {FORMAT_DATA_TYPE_UNORM, FORMAT_ASPECT_COLOR, 4, -1, FORMAT_LAYOUT_COMPRESSED, FORMAT_COMPONENT_RED_GREEN, {{-1, -1, -1, -1}}},
-    // FORMAT_BC5_SNORM,
-    {FORMAT_DATA_TYPE_SNORM, FORMAT_ASPECT_COLOR, 4, -1, FORMAT_LAYOUT_COMPRESSED, FORMAT_COMPONENT_RED_GREEN, {{-1, -1, -1, -1}}},
-    // FORMAT_BC6H_UFLOAT,
-    {FORMAT_DATA_TYPE_FLOAT, FORMAT_ASPECT_COLOR, 4, -1, FORMAT_LAYOUT_COMPRESSED, FORMAT_COMPONENT_RED_GREEN_BLUE, {{-1, -1, -1, -1}}},
-    // FORMAT_BC6H_SFLOAT,
-    {FORMAT_DATA_TYPE_FLOAT, FORMAT_ASPECT_COLOR, 4, -1, FORMAT_LAYOUT_COMPRESSED, FORMAT_COMPONENT_RED_GREEN_BLUE, {{-1, -1, -1, -1}}},
-    // FORMAT_BC7_UNORM,
-    {FORMAT_DATA_TYPE_UNORM, FORMAT_ASPECT_COLOR, 4, -1, FORMAT_LAYOUT_COMPRESSED, FORMAT_COMPONENT_RED_GREEN_BLUE_ALPHA, {{-1, -1, -1, -1}}},
-    // FORMAT_BC7_SRGB,
-    {FORMAT_DATA_TYPE_SRGB, FORMAT_ASPECT_COLOR, 4, -1, FORMAT_LAYOUT_COMPRESSED, FORMAT_COMPONENT_RED_GREEN_BLUE_ALPHA, {{-1, -1, -1, -1}}},
+    // We don't support non-square blocks for compressed textures.
+    //                +-----------------------------------------------------------+
+    //                |                             ,-> bytes per block           |
+    //                |                             |  ,-> block width            |
+    //                | grfx format       | type    |  |   Components             |
+    COMPRESSED_FORMAT(BC1_RGBA_SRGB,        SRGB,   8, 4,  RED_GREEN_BLUE_ALPHA),
+    COMPRESSED_FORMAT(BC1_RGBA_UNORM,       UNORM,  8, 4,  RED_GREEN_BLUE_ALPHA),
+    COMPRESSED_FORMAT(FORMAT_BC1_RGB_SRGB,  SRGB,   8, 4,  RED_GREEN_BLUE),
```

Comment:
Oh so that was a mistake.
Since it's an array, the index is important, and before there was this comment to say: this line is this grfx::FORMAT_X.
The first argument of the define here is not used. It's purely an hint for reading this table.
So made the mistake of not removing the redundant FORMAT_ prefix for this hint block.

Looking back, it was a bad idea to make it look like an argument. Changed that to a comment.